vmdb/app/models/host_openstack_infra.rb

```diff
@@ -1,2 +1,5 @@
 class HostOpenstackInfra < Host
+  def self.hostname_required?
+    false
+  end
```

Comment:
I don't know, having the validation gets in the way of host provisioning also, see commit 2.  I'd love to remove it all if it's not required.

So, the initial 4dd2d13fc0a3 SVN History merged commit had: 
`validates_presence_of     :name, :hostname, :ipaddress, :vmm_vendor`

You changed it to only require hostname/ipaddress if it's vmware to add amazon ec2 support:

```
commit 79b0e8df33f62afcd9b336f78ff1efb8127f4236
Author: Jason Frey <jfrey@redhat.com>
Date:   Mon Feb 2 23:28:42 2009 +0000

    Added initial support for Amazon EC2.
    BugzID: 4776
```